change/@ni-nimble-components-af9b983d-43db-46f7-a204-c271899c2559.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "major",
```

Comment:
Chatted with meyers, some high level things to look into:
- Can DataHierarchyManager be the owner of the data (i.e. the constructor receives the data, calculates isDataFlat). May mean the validator needs to update to understand DataHierarchyManager instead of data references.
- Similar to selection manager, can the expansion manager own configuring expansion state on TanStackTable (right now the top-level table owns updating both separately). Can we document why we need to persist collapseRows state outside of tanstack today and is there a path to tanstack completely owning that state (we submit a bug / PR, doesn't block this PR but we work towards it)
- Can we use generics on arrayToTree to more strongly type the config, i.e. each record must have fields for id and parentid that are keyof the generic

Reply:
So here are the changes (or non-changes) that I have made for each of the mentioned issues/concerns and some accompanying reasoning:
- Went ahead and refactored the `DataHierarchyManager` as discussed. Nothing in the validator needed to be changed.
- Added documentation to top of `ExpansionManager` class to outline the reasons we need to maintain separate state from TanStack. Added a reference to `TanStackTable` on the `ExpansionManager` which allows it call appropriate functions directly like `toggleAllRowsExpanded`.
    - I'm not sure there is a good path to request the kinds of behaviors from TanStack that we are implementing for expanded state. For now, I think our auxiliary state is fairly easy to reason about and hooking it into the TanStack pipeline is fairly trivial.
- Made `arrayToTree` generic, strongly-typed, and less configurable (allowing us to remove a swath of tests we didn't care about).